### PR TITLE
[Merged by Bors] - feat: port/CategoryTheory.PEmpty

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2326

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2326
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2326~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by mattrobball** [comment]
Unblocked by `DiscreteCategory` and should be good to go. 

Note: this file, despite its length, puts serious strain on Lean

File: Mathlib/CategoryTheory/Pempty.lean [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+/-!
+# The empty category
+
+Defines a category structure on `PEmpty`, and the unique functor `PEmpty ⥤ C` for any category `C`.
+-/
+
+
+universe w v u
+
+-- morphism levels before object levels. See note [CategoryTheory universes].
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Pempty.lean:22]:
```suggestion
universe w v u
-- morphism levels before object levels. See note [CategoryTheory universes].

```

**Review by jcommelin — COMMENTED** [review]
Could you please leave a porting note in the file about the strain placed on Lean?

bors d+

File: Mathlib/CategoryTheory/Pempty.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Pempty.lean]: Done.

**Comment by mattrobball** [comment]
> Could you please leave a porting note in the file about the strain placed on Lean?

Added a note at the top. Thought I’ve since found a worse one 😧

**Pull request closed without merging**